src/GraphQL.MicrosoftDI.Tests/ScopedAttributeTests.cs

```diff
@@ -0,0 +1,96 @@
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI.Tests
+{
+    public class ScopedAttributeTests
+    {
+        [Fact]
+        public async void ScopedMethodWorks()
+        {
+            Class1.DisposedCount = 0;
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddScoped<Class1>();
+            serviceCollection.AddScoped<Class2>();
+            var rootServiceProvider = serviceCollection.BuildServiceProvider(false);
+            var graphType = new AutoRegisteringObjectGraphType<TestClass>();
+            var context = new ResolveFieldContext
+            {
+                Source = new TestClass(),
+                RequestServices = rootServiceProvider,
+            };
+            var unscopedFieldResolver = graphType.Fields.Find(nameof(TestClass.UnscopedField))!.Resolver!;
+            var scopedFieldResolver = graphType.Fields.Find(nameof(TestClass.ScopedField))!.Resolver!;
+            var scopedAsyncFieldResolver = graphType.Fields.Find(nameof(TestClass.ScopedAsyncField))!.Resolver!;
+            unscopedFieldResolver.Resolve(context).ShouldBe("0 1");
+            unscopedFieldResolver.Resolve(context).ShouldBe("1 2");
+            unscopedFieldResolver.Resolve(context).ShouldBe("2 3");
+            Class1.DisposedCount.ShouldBe(0);
+            (await (Task<object>)scopedFieldResolver.Resolve(context)).ShouldBe("0 1");
+            Class1.DisposedCount.ShouldBe(1);
+            (await (Task<object>)scopedFieldResolver.Resolve(context)).ShouldBe("0 1");
+            Class1.DisposedCount.ShouldBe(2);
+            (await (Task<object>)scopedAsyncFieldResolver.Resolve(context)).ShouldBe("0 1");
+            Class1.DisposedCount.ShouldBe(3);
+            (await (Task<object>)scopedAsyncFieldResolver.Resolve(context)).ShouldBe("0 1");
+            Class1.DisposedCount.ShouldBe(4);
+            unscopedFieldResolver.Resolve(context).ShouldBe("3 4");
+            rootServiceProvider.Dispose();
+            Class1.DisposedCount.ShouldBe(5);
+        }
+
+        private class TestClass
+        {
+            public string UnscopedField([FromServices] Class1 arg1, [FromServices] Class2 arg2)
```

Comment:
By the way, how to deal with a such (awkward) cases:
```
  [Scoped]
  public string BlaBla([FromServices] Class1 arg1, [FromServices] Class1 arg2,  [FromServices] Class1 arg3,  [FromServices] Class1 arg4)
```

Would be arg1..4 different references or the same?

Reply:
I think the same.

Reply:
It would depend on if they were registered as transients (different) or scoped/singletons (same), exactly like if you did the same in a constructor.

Reply:
ok